File: src/main/java/ru/citeck/ecos/codeinsight/forms/components/Columns.java
package ru.citeck.ecos.codeinsight.forms.components;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
public class Columns extends Component {

    public static final String TYPE = "columns";

    private final List<Column> columns = new ArrayList<>();
    private final Boolean oneColumnInViewMode = true;

    public Columns(Integer columnsCount) {
        for (int i = 0; i < columnsCount; i++) {
            Column column = new Column();
            column.setIndex(i);
            this.columns.add(column);
        }
    }

    @Override
    public String getType() {
        return TYPE;
    }

    public Boolean getOneColumnInViewMode() {
        return oneColumnInViewMode;
    }

    public List<Column> getColumns() {
        return columns;
    }

    @Override
    @JsonIgnore
    public String getLabel() {
        return super.getLabel();
    }
}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/codeinsight/forms/components/Columns.kt
package ru.citeck.idea.artifacts.action.codeinsight.forms.components

import ecos.com.fasterxml.jackson210.annotation.JsonIgnore

class Columns(columnsCount: Int) : ru.citeck.idea.artifacts.action.codeinsight.forms.components.Component() {

    companion object {
        const val TYPE: String = "columns"
    }

    val columns: MutableList<ru.citeck.idea.artifacts.action.codeinsight.forms.components.Column> = ArrayList()
    val oneColumnInViewMode: Boolean = true

    init {
        for (i in 0..<columnsCount) {
            val column = ru.citeck.idea.artifacts.action.codeinsight.forms.components.Column()
            column.index = i
            columns.add(column)
        }
    }

    override fun getType(): String {
        return ru.citeck.idea.artifacts.action.codeinsight.forms.components.Columns.Companion.TYPE
    }

    @JsonIgnore
    override fun getLabel(): String {
        return super.getLabel()
    }
}
